rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -152,6 +190,27 @@ def _decode_curve_events(  # pylint: disable=no-self-use
 
         return None, None
 
+    def _maybe_enrich_curve_transfers(  # pylint: disable=no-self-use
+            self,
+            token: EthereumToken,  # pylint: disable=unused-argument
+            tx_log: EthereumTxReceiptLog,
+            transaction: EthereumTransaction,
+            event: HistoryBaseEntry,
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+    ) -> bool:
+        source_address = hex_or_bytes_to_address(tx_log.topics[1])
+        to_address = hex_or_bytes_to_address(tx_log.topics[2])
+        if (  # deposit give asset
+            event.event_type == HistoryEventType.RECEIVE and
+            event.event_subtype == HistoryEventSubType.NONE and
+            source_address == CURVE_Y_DEPOSIT and
+            transaction.from_address == to_address
+        ):
+            event.event_subtype = HistoryEventSubType.REMOVE_ASSET
```

Comment:
Okay so here we may have a problem.

We need to figure out what we want to call deposit and withdrawal in pools.

As you can see we treat it the same in say aave, compound.

So deposit is:
```
                event.event_type = HistoryEventType.DEPOSIT
                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
```

Receive the wrapped/pool token is:

```
                event.event_type = HistoryEventType.RECEIVE
                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
```


Then withdrawing from the pool is. Step 1(send back pool token):

```
                event.event_type = HistoryEventType.SPEND
                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
```

step 2: get back the actual token:

```
                event.event_type = HistoryEventType.WITHDRAWAL
                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
```

This is how aave/compound pools are.

You are doing it different here. This is why they don't show well in the frontend. There is no mapping for this.
The mappings of all possible combos are here:

https://github.com/rotki/rotki/blob/4e50a76b62362714e589b2bebf8370d5a6e24295/frontend/app/src/store/history/consts.ts#L289

I would say it still makes sense to treat them like we do with aave and compound? Right?

Reply:
Absolutely. I've fixed that one and checked in the frontend